File: Data Structures & Algorithms/binary-tree-right-side-view/submission-2.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[]}
     */
    rightSideView(root) {
        const res = [];
        const queue = [];
        if (root){
            queue.push(root);
        }
        while (queue.length){
            
            let length = queue.length;
            res.push(queue[0].val);
            for (let i = 0; i < length; i++){
                let curr = queue.shift();
                if (curr.right){
                    queue.push(curr.right);
                }
                if (curr.left){
                    queue.push(curr.left);
                }
            }
        }

        return res;
    }
}

File: Data Structures & Algorithms/invert-a-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        
        self.invertSubTree(root)
        return root
    def invertSubTree(self, node):
        if not node:
            return 

        if not node.left and not node.right:
            return node

        temp = node.left
        node.left = node.right
        node.right = temp

        if node.left:
            self.invertTree(node.left)
        if node.right:
            self.invertTree(node.right)
        